experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -793,6 +811,16 @@ export class ModularChangeFamily
 			}
 		}
 
+		// If there's a node exists constraint and we deleted the node, increment the violation count
+		if (rebasedChange.nodeExistsConstraint !== undefined && deleted === true) {
+			// Only increment the violation count if the constraint wasn't already violated
```

Comment:
Shouldn't `deleted === true` already guarantee that the "node exists" constraint wasn't violated before (since you can't delete an already deleted node)?
FWIW, I like the extra check here, I just want to check my understanding.

Reply:
I'm not actually sure what the guarantees around node deletion are (and why I wrote the check). I wrote this with concurrent deletes in mind because I wasn't sure how they're handled and want to avoid a "double increment" problem.